flocker/control/test/test_httpapi.py

```diff
@@ -451,10 +451,146 @@ def get_dataset_ids(deployment):
+        """
+        saving = self.persistence_service.save(deployment)
+
+        def saved(ignored):
+            return self.assertResultItems(
+                b"GET", b"/datasets", None, OK, expected
+            )
+        saving.addCallback(saved)
+        return saving
+
+    def test_one_dataset(self):
+        """
+        When the cluster configuration includes one dataset, the
+        endpoint returns a single-element list containing the dataset.
+        """
+        manifestation = self._manifestation()
+        deployment = Deployment(
+            nodes={
+                Node(
+                    hostname=self.NODE_A,
+                    other_manifestations=frozenset({manifestation}),
+                ),
+            },
+        )
+        expected = [
+            api_dataset_from_dataset_and_node(
+                manifestation.dataset, self.NODE_A
+            )
+        ]
+        return self._dataset_test(deployment, expected)
+
+    def test_several_nodes(self):
+        """
+        When the cluster configuration includes several nodes, each
+        with a dataset, the endpoint returns a list containing
+        information for the dataset on each node.
+        """
+        manifestation_a = self._manifestation()
+        manifestation_b = self._manifestation()
+        deployment = Deployment(
+            nodes={
+                Node(
+                    hostname=self.NODE_A,
+                    other_manifestations=frozenset({manifestation_a}),
+                ),
+                Node(
+                    hostname=self.NODE_B,
+                    other_manifestations=frozenset({manifestation_b}),
+                ),
+            },
+        )
+        expected = [
+            api_dataset_from_dataset_and_node(
+                manifestation_a.dataset, self.NODE_A
+            ),
+            api_dataset_from_dataset_and_node(
+                manifestation_b.dataset, self.NODE_B
+            ),
+        ]
+        return self._dataset_test(deployment, expected)
+
+    def test_several_datasets(self):
+        """
+        When the cluster configuration includes a node with several
+        datasets, the endpoint returns a list containing information
+        for each dataset.
+        """
+        manifestation_a = self._manifestation()
+        manifestation_b = self._manifestation()
+        deployment = Deployment(
+            nodes={
+                Node(
+                    hostname=self.NODE_A,
+                    other_manifestations=frozenset({
+                        manifestation_a, manifestation_b
+                    }),
+                ),
+            },
+        )
+        expected = [
+            api_dataset_from_dataset_and_node(
+                manifestation_a.dataset, self.NODE_A
+            ),
+            api_dataset_from_dataset_and_node(
+                manifestation_b.dataset, self.NODE_A
+            ),
+        ]
+        return self._dataset_test(deployment, expected)
+
```

Comment:
Maybe add unit test for dataset that has metadata, so we can see that the schema verification doesn't reject it.